src/components/Toggle.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+
+import "../stylesheets/ToggleSwitch.scss"
+
+
+/**
+ * Toggle Switch State
+ * checked: true if on, false otherwise
+ */
+export interface ToggleState {
+    checked: boolean;
+}
+
+/**
+ * Toggle Switch Props
+ * defaultChecked: default state at init.
+ */
+export interface ToggleProps {
+    defaultChecked: boolean;
+}
+
+/**
+ * A Toggle Switch component configured with ToggleProps
+ */
+export class ToggleSwitch extends React.Component<ToggleProps, ToggleState> {
+    public static defaultProps: Partial<ToggleProps> = {
+        defaultChecked: false
+    };
+
+    public readonly state: Readonly<ToggleState> = {
+        checked: this.props.defaultChecked
+    };
+
+    /**
+    * Toggle on and off the switch.
+    */
+    toggle() {
+        this.setState({ checked: !this.state.checked })
```

Comment:
I think we need to this instead:
```
this.setState((state) => { checked: !state.checked })
```
Since we're using the previous state to update to next state. Some async stuff i don't really recall LOL.

Reply:
It doesn't compile :smileyface:

Reply:
Probably missing a bracket or something